thoth/adviser/predictors/latest.py

```diff
@@ -41,24 +42,20 @@ class ApproximatingLatest(HillClimbing):
     if resolution to all latest cannot be satisfied.
     """
 
-    _hop = attr.ib(type=bool, default=False)
-    _hop_logged = attr.ib(type=bool, default=False)
-    _packages_heated_up = attr.ib(type=Set[str], factory=set)
-    _initial_state = attr.ib(type=Optional[State], default=None)
-    _latest_versions_heat_up = attr.ib(type=set, factory=set)
+    _hop = attr.ib(type=bool, default=False, init=False)
+    _hop_logged = attr.ib(type=bool, default=False, init=False)
+    _packages_heated_up = attr.ib(type=Set[str], factory=set, init=False)
+    _initial_state = attr.ib(type=Optional[State], default=None, init=False)
+    _latest_versions_heat_up = attr.ib(type=Set[Any], factory=set, init=False)
```

Comment:
Why `init=False` here and other places?

Reply:
Because it shouldn't be a keyword or positional argument. They are only set by internal functions